application/src/main/java/uk/gov/hmcts/ccd/definition/store/ApplicationConfiguration.java

```diff
@@ -1,15 +1,30 @@
 package uk.gov.hmcts.ccd.definition.store;
 
+import java.util.concurrent.Executor;
+
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
 import org.springframework.web.client.RestTemplate;
 
 @Configuration
+@EnableAsync
 class ApplicationConfiguration {
 
     @Bean
     RestTemplate restTemplate() {
         return new RestTemplate();
     }
 
+    @Bean
+    public Executor asyncExecutor() {
+        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+        executor.setCorePoolSize(10);
+        executor.setMaxPoolSize(10);
+        executor.setQueueCapacity(500);
+        executor.setThreadNamePrefix("async-exec-");
```

Comment:
Should the variables come from properties file?

Reply:
@adr1ancho I haven't issued  an official PR for this yet. This is not ready yet.